doc: add #careers vetting criteria and link to README (#8954)
kubernetes/community: communication/slack-job-posting-policy.md

- **Complete information:** Job posts must include a meaningful description and link to an official career portal. "DM for details" or submissions with insufficient information will be rejected.
- **Individual resumes only:** Community members may post personal resumes; however, third-party recruitment services, talent farming, and bulk resume dumps are not permitted.

## 2. Rejection Flags:

Posts matching these criteria are rejected:

- **Security threats:** URL shorteners (such as bit.ly, tinyurl) that mask destinations; links to known phishing or typosquatting domains, or suspicious links of any kind.
- **Spam or bot activity:** Automated posts, generic mass-posting templates, or identical posts repeated across channels or time periods.
- **Off-topic content:** Cryptocurrency mining, generic "work from home" virtual assistant positions, or roles with no technical connection to Kubernetes or cloud-native technologies.
- **Incomplete submissions:** Posts missing critical details such as company name, job title, or clear application instructions.

## 3. Gray Areas:

> Some submissions fall into ambiguous categories and require moderator judgment:

Reviewer: ```suggestion
Some submissions fall into ambiguous categories and require moderator judgment.  Generally this will mean removing the post and messaging the poster to clarify their intent or identity.
```

> - **Borderline relevance:** Roles that are adjacent to cloud-native work (such as general Linux administration without explicit Kubernetes mention) but may still interest community members.

Reviewer: Hmmm.  I don't know that we want to encourage this by suggesting that it's possible.  Folks will push boundaries on their own.

- **Emerging companies:** Early-stage startups with limited online presence may require verification (LinkedIn profile, domain age, or other verification signals) to confirm legitimacy.
- **High-volume posters:** Legitimate recruiting agencies and firms that post frequently may need guidance to avoid cluttering the channel.

## 4. Security & Enforcement:

Submissions identified as malicious will be handled with immediate enforcement:

> - **Identity theft, phishing, or scam attempts:** Posts will be deleted immediately, and the user will be banned from the Slack workspace. **No appeals are permitted for security violations.**

Reviewer: I don't think we need to be this severe.  It's going to be rare for us to *know* that a post was actually a scam attempt.  If they can figure out how to file an appeal via a GH issue, we'd reconsider.  However, to deter AI agents, we won't explain exactly how to do that:

```suggestion
- **Identity theft, phishing, or scam attempts:** Posts will be deleted immediately, and the user will be banned from the Slack workspace.
```

## 5. Feedback & Appeals

We maintain transparency and fairness in our moderation process:

- **Rejection notifications:** Users whose posts are rejected receive a clear explanation citing the specific criteria that resulted in rejection.

> - **Appeal process:** Users who believe a post was rejected in error may reach out to `#slack-admins` for manual review. Security & Enforcement rejections are final and not subject to appeal.

Reviewer: ```suggestion
- **Appeal process:** Users who believe a post was rejected in error may reach out to `#slack-admins` for manual review.  If the rejection resulted in a ban, the user will need to file an issue in `kubernetes/community`.
```
